core/src/main/scala/kafka/utils/Json.scala

```diff
@@ -30,17 +32,26 @@ object Json {
   private val mapper = new ObjectMapper()
 
   /**
-   * Parse a JSON string into a JsonValue if possible. `None` is returned if `input` is not valid JSON.
-   */
+    * Parse a JSON string into a JsonValue if possible. `None` is returned if `input` is not valid JSON.
+    */
   def parseFull(input: String): Option[JsonValue] =
+    try Option(mapper.readTree(input)).map(JsonValue(_))
+    catch { case _: JsonProcessingException => None }
+
+
+  /**
+   * Parse a JSON string into a JsonValue if possible. `None` is returned if `input` is not valid JSON. This method is currently used
+    * to read the already stored invalid ACLs JSON which was persisted using older versions of Kafka (prior to Kafka 1.1.0). KAFKA-6319
+   */
+  def parseBytesIncludingACLs(input: Array[Byte]): Option[JsonValue] =
     try Option(mapper.readTree(input)).map(JsonValue(_))
     catch {
       case _: JsonProcessingException =>
         // Before 1.0.1, Json#encode did not escape backslash or any other special characters. SSL principals
         // stored in ACLs may contain backslash as an escape char, making the JSON generated in earlier versions invalid.
         // Escape backslash and retry to handle these strings which may have been persisted in ZK.
         // Note that this does not handle all special characters (e.g. non-escaped double quotes are not supported)
-        val escapedInput = input.replaceAll("\\\\", "\\\\\\\\")
+        val escapedInput = new String(input, StandardCharsets.UTF_8).replaceAll("\\\\", "\\\\\\\\")
         try Option(mapper.readTree(escapedInput)).map(JsonValue(_))
         catch { case _: JsonProcessingException => None }
```

Comment:
Could we use `parseBytes(escapedInput)` instead of the above two lines?

Reply:
Good point. I'll update the PR soon.